pkg/clients/es.go

```diff
@@ -185,45 +234,67 @@ func doIndex(ctx context.Context, index, id string, doc interface{}) (*esapi.Res
 		esClient.Index.WithDocumentID(id),
 	)
 	if err != nil {
-		logrus.Error(err)
+		logrus.Errorf("Index request failed: %v", err)
 		return nil, fmt.Errorf("index request failed: %w", err)
 	}
 
 	if res.IsError() {
-		logrus.Error(res.String())
+		logrus.Errorf("Index returned error: %s", res.String())
 		return res, fmt.Errorf("index returned error: %s", res.String())
 	}
 
+	logrus.Infof("Document %s indexed successfully in %s", id, index)
 	return res, nil
 }
 
-// =====================
-// PUBLIC FUNCTIONS
-// =====================
-
-// EnqueueSearchTask
-// shouldn't be used due to latency, use doSearch
-//func EnqueueSearchTask(ctx context.Context, index string, query interface{}) <-chan *SearchResult {
-//	responseChan := make(chan *SearchResult, 1)
-//	req := SearchRequest{
-//		Ctx:          ctx,
-//		Index:        index,
-//		Query:        query,
-//		ResponseChan: responseChan,
-//	}
-//	taskQueueSearch <- req
-//	return responseChan
-//}
-
-func EnqueueIndexTask(ctx context.Context, index, id string, document interface{}) <-chan *IndexResult {
+func EnqueueIndexTask(ctx context.Context, index, id string, document interface{}, function _const.Function) <-chan *IndexResult {
```

Comment:
All the functions that call this function don't use the returned value...
Should they?

Reply:
No, also `fire-and-forget` is the best approach in this scenario.